visualization/app/codeCharta/state/store/fileSettings/blacklist/blacklist.reducer.ts

```diff
@@ -10,7 +9,7 @@ export function blacklist(state: BlacklistItem[] = setBlacklist().payload, actio
 		case BlacklistActions.REMOVE_BLACKLIST_ITEM:
 			return removeItemFromArray(state, action.payload)
 		case BlacklistActions.SET_BLACKLIST:
-			return _.cloneDeep(action.payload)
+			return [...action.payload]
```

Comment:
This results in a shallow clone. Is that intentional?

Reply:
This is intentional. redux uses immutable update patterns for performance. 

https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns/

Reply:
Sorry, I can't follow. The payload does not seem to be changed in this function. Is the shallow clone still necessary? I would expect no cloning to be necessary?

Reply:
Redux uses pure functions as reducers. Therefore the state object itself should not be mutated under any circumstances and instead return a new state.

Not shallow copying the previous state would result in mutating it and breaking the pure function. We use a shallow copy of the previous state and simply add the new item aka the payload there.

Reply:
Oh god, ignore my previous comment. I thought this was the ADD_BLACKLIST_ITEM action.

Shallowcopying the blacklist here doesn't make any sense. It may make sense in cloning it though as done previously (see comment below)

This change needs to be reverted.